packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue

```diff
@@ -290,30 +289,28 @@ async function buildJupyterContext() {
 	}
 }
 
+// Generate once the comparison task has been completed
+function generateOverview(output: string) {
+	overview.value = markdownit().render(JSON.parse(b64DecodeUnicode(output)).response);
+	emit('update-status', OperatorStatus.DEFAULT); // This is a custom way of granting a default status to the operator, since it has no output
+}
+
+// Create a task to compare the models
 async function processCompareModels(modelIds: string[]) {
 	const taskRes = await compareModels(modelIds, props.node.workflowId, props.node.id);
 	compareModelsTaskId = taskRes.id;
 	if (taskRes.status === TaskStatus.Success) {
-		compareModelsTaskOutput = taskRes.output;
+		generateOverview(taskRes.output);
 	}
 }
 
-function assignOverview(b64overview: string) {
-	overview.value = markdownit().render(JSON.parse(b64DecodeUnicode(b64overview)).response);
-}
-
-async function generateOverview() {
-	// Generate once the comparison task has been completed
-	if (!compareModelsTaskOutput) return;
-	assignOverview(compareModelsTaskOutput);
-	emit('update-status', OperatorStatus.DEFAULT); // This is a custom way of granting a default status to the operator, since it has no output
-}
-
+// Listen for the task completion event
 useClientEvent(ClientEventType.TaskGollmCompareModel, (event: ClientEvent<TaskResponse>) => {
-	if (!event.data || event.data.id !== compareModelsTaskId) return;
+	if (!isEmpty(overview.value) && event?.data.id === compareModelsTaskId) return;
```

Comment:
Just curious, are there cases where an event with a takeId is fired after an overview has already been generated from the task output with the same taskId?

Reply:
Nope, there is not, but there is a situation were the request to get an overview is answered before the client-event does and therefore there is no need to overwrite it twice.